mechanicalsoup/browser.py

```diff
@@ -240,7 +240,14 @@ def __bool__(self):
 
             files = DictThatReturnsTrue()
 
-        return self.session.request(method, url, files=files, **kwargs)
+        request_kwargs = {"method": method, "url": url, "files": files}
+        request_kwargs.update(kwargs)
```

Comment:
If the user provides `method` or `url` arguments to `**kwargs` in our API, it would currently raise a `TypeError` due to a repeated argument passed to `session.request` (I'm working on a unit test to demonstrate this now). However, in this patch, those keys would be overwritten by the `request_kwargs` dict update. It seems like it would be a little clunky to manually check for those keys, but maybe that's what's needed here to retain the expected behavior?

Reply:
I hadn't thought about TypeError. Thank you for pointing that out.

It looks like we can check the key for backwards compatibility. There seems to be more than one way to implement it.

Which of the following implementation methods do you prefer?

(a) Raise TypeError in the package
(b) Or create a class method, e.g. `_get_request_kwargs()` that only merges the kwargs so that Python raises a TypeError.

Reply:
Since we have difficulty thinking about the msg of TypeError(msg), I thought it might be more desirable for Python to raise a TypeError.

Reply:
536ce4a398c88a3d4bdf26da19502619ebb2de67 Resolve this issue.